Document proposed decision around cflinuxfs3 to 4 change (#475)
alphagov/paas-team-manual: source/architecture_decision_records/ADR051-cflinuxfs3-to-cflinuxfs4-migration.html.md

  on the platform and the variety of technologies in use.
  <br /><br />

* Tenants are often slow to respond to communications and calls to action. This is a 
  problem we have faced through the lifetime of the platform, and we usually build this 
  into our plans by providing a long window (for example 9 months for Postgres 10 deprecation)
  in which to act and issuing regular reminders. Unfortunately in this case we have not 
  given ourselves a long lead time, and as of 14th Feb 2023 the CloudFoundry Foundation 
  still has 3 buildpacks missing support for `cflinuxfs4`.
  <br /><br />
  Regardless, we must provide an adequate timeframe for tenants to act in.
  <br /><br />

* GOV.UK PaaS is in the middle of being decommissioned, and every tenant is being asked
  to migrate away from the platform. For many of them this is consuming a significant
  proportion of their development capacity, and in most cases they will have to work out
  how to prioritize migration activities against the work to swap to `cflinuxfs4`.
  <br /><br />
  We must set our expectations of how quickly tenants will act accordingly. 

>   Some may choose not to migrate at all, in favour of trying to migrate away before any 

Reviewer: ```suggestion
  Some may choose not to transition to `cflinuxfs4`, in favour of trying to migrate away from the platform before any 
```
Is this what you mean? the double migration confused me

   who cannot, or choose not to, make the switch. Tenants who do this will be doing so
   at their own risk, and must clear it with their own Information Assurance teams, 
   risk management processes etc.

### Risk mitigations/controls
Basing tenant applications on an unsupported OS version presents us a medium level risk, due
to the opportunity for vulnerabilities of varying severity to go un-patched, and exploitation to go
unnoticed. To account for this we will monitor [the Ubuntu Security Notices for Bionic](https://ubuntu.com/security/notices?order=newest&release=bionic&details=)
so that we have continued visibility of any vulnerabilities. On notification of a CVE the team will 
be responsible for understanding its scope and severity. We will inform tenants as necessary, and 
point them to mitigations they can put in place.

We mitigate the risk somewhat by reducing the number of instances in which a container is running
with `cflinuxfs3` as the base. We can do this by encouraging tenants to move to `cflinuxfs4`, and 
making it clear to them that if, at the end of the 6-month period, they are still using `cflinuxfs3`

> then they are responsible for the risk to their own applications.

Reviewer: Just to confirm, between April and when the 6 months end, we share the risk? So it's only from the end of those 6 months onwards that tenants "will be doing so at their own risk, and must clear it with their own Information Assurance teams, risk management processes etc."?

Reviewer: ah, I've just read the bit under Consequences and this makes sense

## Status

Draft

## Consequences

As a consequence of the above decision, we will be basing tenant applications on an unsupported

> operating system version, by default, for 4 months. This is an elevated risk that we accept,

Reviewer: ```suggestion
operating system version, by default, for 4 months (June to September). This is an elevated risk that we accept,
```
Are those the months?

Author: No, it'll likely be April-July

Reviewer: ah yes, I think this was leftover from when it was a 6 months thing
